httputil/request.go

```diff
@@ -172,14 +180,19 @@ func SetAuthType(req *http.Request, res *http.Response) {
 }
 
 func GetAuthType(res *http.Response) string {
-	auth := res.Header.Get("Www-Authenticate")
-	if len(auth) < 1 {
-		auth = res.Header.Get("Lfs-Authenticate")
-	}
 
-	if strings.HasPrefix(strings.ToLower(auth), "ntlm") {
-		return "ntlm"
+	for _, headerName := range []string{WwwAuthenticateHeader, LfsAuthenticateHeader} {
```

Comment:
The inner contents of this loop look good. I wonder if we could make things a little bit clearer by making removing the `WwwAuthenticateHeader` and `LfsAuthenticateHeader` `consts` in favor of a single:

```go
var (
        AuthenticateHeaders = []string{"Www-Authenticate", "Lfs-Authenticate"}
)
```

since we never use them individually, only in this form on line 184.